ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -238,17 +238,19 @@ def dist_to_tl(self):
     def decelerate(self, waypoints):
         dist = self.dist_to_tl()
         if self.tl_idx > - 1 and dist < LOOKAHEAD_WPS:
-            prev_wp = waypoints[dist]
-            self.set_waypoint_velocity(waypoints, dist, 0.)
-            for i in range(dist-1, -1, -1):
+            prev_idx = max(dist-2, 0)
+            prev_wp = waypoints[prev_idx]
+            for i in range(prev_idx, LOOKAHEAD_WPS):
+                self.set_waypoint_velocity(waypoints, i, 0.)
+
+            for i in range(prev_idx-1, -1, -1):
                 wp = waypoints[i]
-                prev_vel = self.get_waypoint_velocity(prev_wp)
                 d = self.distance(waypoints, i, i + 1)
-                # Approximate time between two waypoints
-                t = max(d / (prev_vel + self.max_dec), .1)
+                prev_vel = self.get_waypoint_velocity(prev_wp)
+                vel = math.sqrt(2 * self.max_dec * d + prev_vel**2)
```

Comment:
It's possible that `vel` becomes larger than `self.max_vel`, you should use the `min`

Reply:
Oh nevermind, I just saw the `break` below